Language-Version-History.md

```diff
@@ -0,0 +1,110 @@
+Features Added in F# Language Versions
+====================
+
+# [F# 1.0](https://docs.microsoft.com/en-us/archive/blogs/dsyme/welcome-to-dons-f-blog)
+
+- Discriminated unions
+- Records
+- Tuples
+- Pattern matching
+- Type abbreviations
+- Structs
+- Signature files
+- Imperative programming
+- Modules (no functors)
+- Nested modules
+- .NET Interoperability
+
+# [F# 1.1](https://docs.microsoft.com/en-us/archive/blogs/dsyme/a-taste-of-whats-new-in-f-1-1)
+
+- Interactive environment
+- Object programming
+- Encapsulation Extensions
+
+# [F# 2.0](https://fsharp.org/specs/language-spec/2.0/FSharpSpec-2.0-April-2012.pdf)
+
+- Active patterns
+- Units of measure
+- Object expressions
+- Sequence expressions
+- Asynchronous programming
+- Agent programming
+- Extension members
+- Named arguments
+- Optional arguments
+- Array slicing
+- Quotations
+- Native interoperability
+- Computation expressions
+
+# [F# 3.0](https://fsharp.org/specs/language-spec/3.0/FSharpSpec-3.0-final.pdf)
+
+- Type providers
+- LINQ query expressions
+- CLIMutable attribute
+- Triple-quoted strings
+- Auto-properties
+- Provided units-of-measure
+
+# [F# 3.1](https://fsharp.org/specs/language-spec/3.1/FSharpSpec-3.1-final.pdf)
+
+- Named union type fields
+- Extensions to array slicing
+- Type inference enhancements
+
+# [F# 4.0](https://fsharp.org/specs/language-spec/4.0/FSharpSpec-4.0-final.pdf)
+
+- `printf` on unitized values
+- Extension property initializers
+- Non-null provided types
+- Primary constructors as functions
+- Static parameters for provided methods
+- `printf` interpolation
+- Extended `#if` grammar
+- Tailcall attribute
```

Comment:
Tailcall attribute?.. Could you please provide a usage example of it?

Reply:
@vasily-kirichenko I dont know much about this one, but found this [link](https://github.com/fsharp/fslang-suggestions/issues/721)

Reply:
That proposal is not even implemented yet.

Reply:
so is better remove it from language features! as I am learning the language now, I dont know all features, so every help is appreciated to add or remove features from the list

Reply:
@leandromoh not sure which source you are using, but the canonical one should be the RFCs listed there https://github.com/fsharp/fslang-design

You can see a folder per language version and a file per feature.

Reply:
@smoothdeveloper there are folders only for version < 4.0. Where area others?